File: app/src/main/java/com/medialink/mysqlite/db/NoteHelper.java
package com.medialink.mysqlite.db;

import android.content.ContentValues;
import android.content.Context;
import android.database.Cursor;
import android.database.SQLException;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;

import com.medialink.mysqlite.DatabaseContract;
import com.medialink.mysqlite.DatabaseHelper;

import static android.provider.BaseColumns._ID;

public class NoteHelper {
    private static final String TABLE_NOTE = DatabaseContract.TABLE_NOTE;
    private static DatabaseHelper dbHelper;
    private static NoteHelper INSTANCE;
    private static SQLiteDatabase database;

    public NoteHelper(Context context) {
        dbHelper = new DatabaseHelper(context);
    }

    public static NoteHelper getInstance(Context context) {
        if (INSTANCE == null) {
            synchronized (SQLiteOpenHelper.class) {
                if (INSTANCE == null) {
                    INSTANCE = new NoteHelper(context);
                }
            }
        }
        return INSTANCE;
    }

    public void open() throws SQLException {
        database = dbHelper.getReadableDatabase();
    }

    public void close() {
        dbHelper.close();
        if (database.isOpen()) {
            database.close();
        }
    }

    public Cursor queryAll() {
        return database.query(
                TABLE_NOTE,
                null,
                null,
                null,
                null,
                null,
                _ID + " ASC"
        );
    }

    public Cursor queryById(String id) {
        return database.query(
                TABLE_NOTE,
                null,
                _ID + " = ?",
                new String[]{id},
                null,
                null,
                null,
                null
        );
    }

    public long insert(ContentValues values) {
        return database.insert(
                TABLE_NOTE,
                null,
                values
        );
    }

    public int update(String id, ContentValues values) {
        return database.update(
                TABLE_NOTE,
                values,
                _ID + " = ?",
                new String[]{id}
        );
    }

    public int deleteById(String id) {
        return database.delete(
                TABLE_NOTE,
                _ID + " = ?",
                new String[]{id}
        );
    }
}

File: MySQLiteKotlin/src/main/java/com/medialink/mysqlitekotlin/db/NoteHelper.kt
package com.medialink.mysqlitekotlin.db

import android.content.ContentValues
import android.content.Context
import android.database.Cursor
import android.database.SQLException
import android.database.sqlite.SQLiteDatabase

class NoteHelper(context: Context) {
    companion object {
        private const val TABLE_NOTE = DatabaseContract.NoteColumns.TABLE_NOTE
        private lateinit var dbHelper: DatabaseHelper
        private var INSTANCE: NoteHelper? = null

        private lateinit var database: SQLiteDatabase

        fun getInstance(context: Context): NoteHelper =
                INSTANCE ?: synchronized(this) {
                    INSTANCE ?: NoteHelper(context)
                }


    }

    init {
        dbHelper = DatabaseHelper(context)
    }

    @Throws(SQLException::class)
    fun open() {
        database = dbHelper.writableDatabase
    }

    fun close() {
        dbHelper.close()
        if (database.isOpen) {
            database.close()
        }

    }

    fun queryAll(): Cursor {
        return database.query(
                TABLE_NOTE,
                null,
                null,
                null,
                null,
                null,
                "${DatabaseContract.NoteColumns._ID} ASC"
        )
    }

    fun queryById(id: String): Cursor {
        return database.query(
                TABLE_NOTE,
                null,
                "${DatabaseContract.NoteColumns._ID} = ?",
                arrayOf(id),
                null,
                null,
                null,
                null
        )

    }

    fun insert(values: ContentValues?): Long {
        return database.insert(TABLE_NOTE, null, values)
    }

    fun update(id: String, values: ContentValues?): Int {
        return database.update(
                TABLE_NOTE,
                values,
                "${DatabaseContract.NoteColumns._ID} = ?",
                arrayOf(id)
        )
    }

    fun deleteById(id: String): Int {
        return database.delete(
                TABLE_NOTE,
                "${DatabaseContract.NoteColumns._ID} = '$id'",
                null
        )
    }

}